pkg/security/admission/admission.go

```diff
@@ -92,38 +91,32 @@ func (c *constraint) Admit(a kadmission.Attributes) error {
 		}
 		matchedConstraints = append(matchedConstraints, saConstraints...)
 	}
-
-	// remove duplicate constraints and sort
 	matchedConstraints = oscc.DeduplicateSecurityContextConstraints(matchedConstraints)
 	sort.Sort(oscc.ByPriority(matchedConstraints))
-
-	providers, errs := oscc.CreateProvidersFromConstraints(a.GetNamespace(), matchedConstraints, c.client)
-	logProviders(pod, providers, errs)
-
-	if len(providers) == 0 {
-		return kadmission.NewForbidden(a, fmt.Errorf("no providers available to validate pod request"))
-	}
-
-	// all containers in a single pod must validate under a single provider or we will reject the request
-	validationErrs := field.ErrorList{}
-	for _, provider := range providers {
-		if errs := oscc.AssignSecurityContext(provider, pod, field.NewPath(fmt.Sprintf("provider %s: ", provider.GetSCCName()))); len(errs) > 0 {
-			validationErrs = append(validationErrs, errs...)
-			continue
-		}
-
-		// the entire pod validated, annotate and accept the pod
-		glog.V(4).Infof("pod %s (generate: %s) validated against provider %s", pod.Name, pod.GenerateName, provider.GetSCCName())
-		if pod.ObjectMeta.Annotations == nil {
-			pod.ObjectMeta.Annotations = map[string]string{}
-		}
-		pod.ObjectMeta.Annotations[allocator.ValidatedSCCAnnotation] = provider.GetSCCName()
-		return nil
+	if err = oscc.AssignConstraints(matchedConstraints, a.GetNamespace(), c.client,
```

Comment:
correct. clarified the comment with an example in https://github.com/openshift/origin/pull/12009#issuecomment-262974785

I think this means reworking (or expanding, not sure) the SCC Validate* interfaces to report two pieces of info - whether the object is valid as-is and whether the corresponding Create* function could make the object valid